Add integration test for server RPM

electrum opened the pull request:
Fixes #1771

findepi commented on presto-server-rpm/pom.xml:
```diff
@@ -169,12 +205,96 @@
+                                                <args>
+                                                    rpm -i /target/${project.build.finalName}.x86_64.rpm
+                                                    /etc/init.d/presto start
+                                                    tail ---disable-inotify -F /var/log/presto/server.log
```
add code comment why `--disable-inotify`

electrum replied:
Done

findepi commented on presto-server-rpm/pom.xml:
```diff
@@ -169,12 +205,96 @@
+                                        </cmd>
+                                        <wait>
+                                            <time>60000</time>
+                                            <log>.*SERVER STARTED.*</log>
```
we could configure hive connector to verify it at least loads
(at startup time, it requires no external services, so it's easy)

electrum replied:
Done

findepi commented on presto-server-rpm/pom.xml:
```diff
@@ -169,12 +205,96 @@
+                <plugins>
+                    <plugin>
+                        <groupId>io.fabric8</groupId>
+                        <artifactId>docker-maven-plugin</artifactId>
```
That's cool, but does it handle docker-compose -style setups?
eg if we wanted to smoke test RPM with a connector?

electrum replied:
Looks like it: https://dmp.fabric8.io/#docker-compose

I was thinking (after I got this all working) that doing this with `DockerContainer` might be better, but this works for now.

findepi commented on presto-server-rpm/src/test/java/io/prestosql/server/rpm/ServerIT.java:
```diff
@@ -0,0 +1,44 @@
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class ServerIT
```
TestServer or TestRpm

electrum replied:
It has to be named this way unless we change the default naming pattern: https://maven.apache.org/surefire/maven-failsafe-plugin/examples/inclusion-exclusion.html

I dislike `IT` but it seems better to use the defaults. Open to suggestions, though.

findepi commented on presto-server-rpm/pom.xml:
```diff
@@ -15,16 +15,51 @@
         <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
```
add comment explaining why we need this

electrum replied:
Done

findepi commented on presto-server-rpm/pom.xml:
```diff
@@ -169,12 +205,96 @@
+                                        <cmd>
+                                            <exec>
+                                                <args>sh</args>
+                                                <args>-c</args>
```
```suggestion
                                                <args>-xeuc</args>
```

electrum replied:
Done

kokosing commented on presto-server-rpm/pom.xml:
```diff
@@ -169,12 +205,96 @@
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>io.fabric8</groupId>
```
This is neat, but I would use `presto-testing-docker` because running `rpm -i` is a part of a test (not setup). However `presto-testing-docker` might be not well prepared to run and validate commands in the container. So change requested.

Additionally with `presto-testing-docker`:
 - we could avoid hardcoding port
 - it would be much easier to run test from the IDE.

electrum replied:
I thought about that as well, but after I already had this working. We can investigate that in the future. The port number only hard-coded inside the container. The host mapping is a random port.

kokosing commented on presto-server-rpm/src/test/java/io/prestosql/server/rpm/ServerIT.java:
```diff
@@ -0,0 +1,44 @@
+        String url = "jdbc:presto://localhost:" + port;
+        try (Connection connection = getConnection(url, "test", null);
+                Statement statement = connection.createStatement();
+                ResultSet rs = statement.executeQuery("SELECT 123 x")) {
```
How about running `SHOW CATALOGS`?

electrum replied:
Done

kokosing commented:
Fixes #1771

electrum commented:
@findepi Thanks, I was thinking it would run as part of `MAVEN_CHECKS`, but that disables tests. I split it up and added a second build command there for the RPM. Not ideal, since it's not a Maven check, but seems like the best place for now (otherwise we'd need a separate job just for it).
